javascript/code-challenges/challenges-06.test.js

```diff
@@ -0,0 +1,179 @@
+{
+  name: 'jamie',
+  age: 21,
+  shoeSize: 8
+}
+]
+
+Returns: ['dyoll', 'eimaj'];
+------------------------------------------------------------------------------------------------ */
+
+const getNames = (arr) => {
+  return arr.map(person => person.name.split('').reverse().join(''));
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function that appends ' The end.' to a string, and returns the modified string. The original source string should not be modified.
+
+------------------------------------------------------------------------------------------------ */
+
+const appendTheEnd = (str) => {
+  return str + ' The end.';
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that accepts an array and copies the first element to the end of the array. The change should be reflected in the source array that was passed in to the function. That is, the function should modify the array 'in place'.
+
+Do not use a return statement.
+
+For example:
+const a = [1, 2, 3];
+appendFirstToLast(a);
+console.log(a) prints [1, 2, 3, 1]
+------------------------------------------------------------------------------------------------ */
+
+const appendFirstToLast = (arr) => {
+  arr.push(arr[0]);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function that accepts an object and an integer as arguments and adds a new property to the object called yearBorn. The value of the yearBorn property should be the integer that was passed in.
+
+The change should be reflected in the source object that was passed in to the function. That is, the function should modify the object 'in place'.
+
+Do not use a return statement.
+
+For example:
+const octavia = { fullName: 'Octavia Estelle Butler' };
+addBirthYearProperty(octavia, 1947);
+console.log(a) prints { fullName: 'Octavia Estelle Butler', yearBorn: 1947 }
+------------------------------------------------------------------------------------------------ */
+
+const addBirthYearProperty = (obj, year) => {
+  obj.yearBorn = year;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5 - Stretch Goal
+
+Write a function that accepts an array of people objects and adds a new property called isAuthor to each object in the list. Set the value of the new property to true.
+
+The function should modify the object in place. Do not use a return statement.
+
+For example:
+const people = [{ fullName: 'Octavia Butler' }, { fullName: 'Ray Bradbury' }];
+setStatusAsAuthor(people);
+console.log(people[1].isAuthor) prints true
+------------------------------------------------------------------------------------------------ */
+
+const setStatusAsAuthor = (people) => {
+  people.map(person => person.isAuthor = true);
+};
+
```

Comment:
Map does not work in place, which is one of the requirements for this problem. So while the tests may be passing, this isn't exactly what we're looking for.
Map will map over the array and then return a new array. So while it may LOOK as if this is happening in place (ie, you console.log your original array and it shows all the objects with the correct properties), what is has actually done is wiped all the data and put in new data that has all of the same values as before. This is slightly different. If you wanted to change everything IN PLACE without actually changing any of the original data inside your argument, you need to do something slightly different.